mockgcp/mockcompute/subnetsv1.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockcompute
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/common/projects"
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/cloud/compute/v1"
+)
+
+type SubnetsV1 struct {
+	*MockService
+	pb.UnimplementedSubnetworksServer
+}
+
+func (s *SubnetsV1) Get(ctx context.Context, req *pb.GetSubnetworkRequest) (*pb.Subnetwork, error) {
+	name, err := s.parseSubnetName("projects/" + req.GetProject() + "/regions/" + req.GetRegion() + "/subnetworks/" + req.GetSubnetwork())
```

Comment:
Silly question but why are we assembling a subnet name here from the 3 component parts, just to parse out those same 3 pieces? I'd get it if this were a public API but its private and all calling sites have the 3 relevant fields...

Reply:
So there is some logic in parseSubnetName around the project.  I'm also trying to normalize the functions across the APIs, hopefully to play into AI.  We are using the normalized name (which we get from `String`) as a key into our "database", so we do want something to normalize the names.

The name parsing also resolves the project (and converts it to an ID for the canonical key), and it could also do things like normalize regions etc in future.

Reply:
Personally I think I would rather have a struct which is the normal way of passing the data around. Internally we should always use the struct and leave the parse function at our API boundary where we can A) easily give an error to the correct caller if there is a problem with the input string and b) not have to worry about parsing and serializing throughout the code base as well as dealing with lack of typing and partial changes to this code.